pkg/compat/client.go

```diff
@@ -13,7 +13,6 @@ import (
 	batchv1beta "k8s.io/api/batch/v1beta1"
 	batchv2alpha "k8s.io/api/batch/v2alpha1"
 	extv1beta1 "k8s.io/api/extensions/v1beta1"
-	"k8s.io/apimachinery/pkg/runtime"
```

Comment:
I think that we can keep this?? 

https://pkg.go.dev/k8s.io/apimachinery/pkg/runtime#Object

Reply:
So the issue seems to be that pretty much across the board, this version of controller-runtime has replaced basically any use of runtime.Object with either client.Object or client.ObjectList.